Update QD to Quality Level 1

brawner opened the pull request:
Signed-off-by: Stephen Brawner <brawner@gmail.com>

brawner committed:
Update QD to Quality Level 1

Signed-off-by: Stephen Brawner <brawner@gmail.com>

wjwwood reviewed (approved):
lgtm

During my review I read the whole document, and though out of the scope of this pr, I found to issues. First the link in this line is 404 for me:

> All the functionality of the declared API in this package is covered in its unit tests. Currently it has a line coverage of [99%](https://ci.ros2.org/job/ci_linux_coverage/85/cobertura/src_ament_ament_index_ament_index_cpp_src/).

Also, this line doesn't make sense to me:

> - achieving and maintaining a reasonable branch line coverage (90-100%)

It should be `branch or line coverage` or just `branch coverage` or just `line coverage`, maybe?

brawner committed:
PR fixup

Signed-off-by: Stephen Brawner <brawner@gmail.com>

brawner commented:
I agree that this PR would be suitable for fixing up any remaining issues. It will get less frequent updates now that it's been bumped to QL 1. I updated the coverage link and updated that line to read `branch or line coverage`.

Blast545 commented on ament_index_cpp/QUALITY_DECLARATION.md:
```diff
@@ -2,7 +2,7 @@ This document is a declaration of software quality for the `ament_index_cpp` pac
 
```
Link to REP-2004 seems to be broken, replace with: [REP-2004 link](https://www.ros.org/reps/rep-2004.html)

brawner replied:
Thanks, fixed.

Blast545 reviewed (approved):
Took a look to the whole document, LGTM after a minor change

brawner committed:
PR Fixup

Signed-off-by: Stephen Brawner <brawner@gmail.com>
